analyses/batch-effects/graph-generator.Rmd

```diff
@@ -0,0 +1,426 @@
+# Batch Effects Graphs Description
+
+This markdown file also exists as a jupyter notebook file for a cleaner version.
+# Library Import & Data Download
+
+#### Import the necessary libararies 
+
+
+```{R}
+library(tibble)
+library(tidyr)
+library(ggplot2)
+library(readr)
+library(dplyr)
+library(BatchQC)
+library(forcats)
+library(sva)
+library(rprojroot)
+```
+
+####  Configure file paths & import functions
+
+
+```{R}
+# Configure file paths
+root_dir = find_root(has_file("OpenPBTA-analysis.Rproj"))
+analysis_dir = file.path(root_dir, "analyses", "batch-effects")
+data_dir = file.path(root_dir, "data")
+functions = file.path(analysis_dir, "util", "functions.R")
+```
+
+
+```{R}
+# Load functions from /util/function.R
+source(functions)
+```
+
+#### Download the Data
+
+
+```{R}
+# Download all covariate data which will be used to identify batches
+covariate_file = file.path(data_dir, "pbta-histologies.tsv")
+covariate = read_tsv(covariate_file, col_types = cols(molecular_subtype = "c"))
+```
+
+
+```{R}
+# Download gene expression data
+dat_rsem_polya_file = file.path(data_dir, "pbta-gene-expression-rsem-tpm.polya.rds")
+dat_rsem_polya = readRDS(dat_rsem_polya_file)
+
+dat_rsem_stranded_file = file.path(data_dir, "pbta-gene-expression-rsem-tpm.stranded.rds")
+dat_rsem_stranded = readRDS(dat_rsem_stranded_file)
+
+dat_kallisto_stranded_file = file.path(data_dir, "pbta-gene-expression-kallisto.stranded.rds")
+dat_kallisto_stranded <- readRDS(dat_kallisto_stranded_file)
+dat_kallisto_stranded = dat_kallisto_stranded[,2:ncol(dat_kallisto_stranded)]
+
+dat_kallisto_polya_file = file.path(data_dir, "pbta-gene-expression-kallisto.polya.rds")
+dat_kallisto_polya <- readRDS(dat_kallisto_polya_file)
+```
+
+##### Shorten the data files if you don't have a strong enough computer 
+
+***This should not be run for a full analysis!***
+
+
+```{R}
+# Shorten Data Files
+dat_rsem_polya = shorten(dat_rsem_polya)
+dat_rsem_stranded = shorten(dat_rsem_stranded)
+dat_kallisto_stranded = shorten(dat_kallisto_stranded)
+dat_kallisto_polya = shorten(dat_kallisto_polya)
+```
+
+#### Group the Kallisto Data
+
+
+```{R}
+# Summarize the kallisto data on the gene level in order to join with batch
+dat_kallisto_polya = grouper(dat_kallisto_polya)
+dat_kallisto_stranded = grouper(dat_kallisto_stranded)
+```
```

Comment:
This collapses to the gene mean, is that correct? Is this step necessary for batch correction?

Reply:
This is not necessary for batch correction. We could apply combat to the transcript level values. However, that is a less proved approach, and people generally sum transcript level values to gene-level values. I had been using the mean as a way of summarization, but Steve Piccolo said that it would be better to summarize using sum. [Here's ](https://github.com/griffithlab/rnaseq_tutorial_wiki/blob/master/Kallisto.md) a tutorial where this is mentioned as well. So I can change that if you're okay with that?